configure.py

```diff
@@ -2415,16 +2405,28 @@ def create_build_system(args):
 
     os.makedirs(outdir, exist_ok=True)
 
+    scylla_product, scylla_version, scylla_release = generate_version(args.date_stamp)
+
+    for mode in modes:
```

Comment:
does it make sense to only update the `build_modes` which is a subset of `mode`. this subset contains the modes which are enabled by user with `--mode` option. this would be more consistent with the change you are adding for querying seastar's exposed building options. also, could use something like,
```py
for mode, config in build_modes.items():
    extra_cxxflags = ' '.join(get_extra_cxxflags(mode, config, args.cxx, args.debuginfo))
    # likewise
```

Reply:
Done.